controllers/seasons.controllers.ts

```diff
@@ -1,7 +1,116 @@
-// TODO: require the model
-// Season = require("../models/Season")
+import { NextFunction, Response, Request } from "express";
+import Season from "../models/Season";
 
-// TODO: add controller logic 
+function create(req: Request, res: Response, next: NextFunction) {
+  const season = new Season({
+    name: req.body.name,
+    startDate: req.body.startDate,
+    endDate: req.body.endDate, //Think this should be calculated when status is changed to close
+    payroll_timeframe: req.body.payroll_timeframe,
+    price: req.body.price,
+    // TODO: Add product_id, unit_id, currency_id
+  });
 
-// TODO: export module
-// module.exports = { controllerName1, controllerName2, etc};
\ No newline at end of file
+  season
+    .save()
+    .then((result) => {
+      res.status(201).json({
+        message: "Season created",
+        result,
+      });
+    })
+    .catch((error) => {
+      res.status(500).json({
+        message: "Season not created",
+        error,
+      });
+    });
+}
+
+function getAll(req: Request, res: Response, next: NextFunction) {
+  Season.find({ deletedAt: null })
+    .select("name status startDate")
+    .exec()
+    .then((results) => {
+      res.status(200).json(results);
+    })
+    .catch((error) => {
+      res.status(500).json(error);
+    });
+}
+
+function getById(req: Request, res: Response, next: NextFunction) {
+  const id = req.params.id;
+
+  Season.findById(id)
+    .exec()
+    .then((results) => {
+      res.status(200).json(results);
+    })
+    .catch((error) => {
+      res.status(500).json(error);
+    });
+}
+
+function close(req: Request, res: Response, next: NextFunction) {
+  const id = req.params.id;
+
+  Season.findOneAndUpdate({ _id: id }, { status: "Closed" })
+    .exec()
+    .then((results) => {
+      res.status(200).json(results);
+    })
+    .catch((error) => {
+      res.status(500).json(error);
+    });
+}
+
+function update(req: Request, res: Response, next: NextFunction) {
+  const id = req.params.id;
+
+  Season.findOneAndUpdate({ _id: id }, req.body)
```

Comment:
findOneAndUpdate as it returns the object without the changes

For example here I'm updating the name, but the result is returning the prev name

<img width="566" alt="Screenshot 2024-02-19 at 9 11 01 AM" src="https://github.com/LangaraVoyagers/project-2-api/assets/57273227/b6a6f6f0-89ec-4d71-8355-34db490dcfef">

When I check with the get by id, the name is changed properly:
<img width="575" alt="Screenshot 2024-02-19 at 9 11 07 AM" src="https://github.com/LangaraVoyagers/project-2-api/assets/57273227/302addbd-f237-4a03-b707-421285cb82fe">

but we need to return the updated values as well,

this can be achieved by adding the option returnDocument:"after"
https://www.mongodb.com/docs/manual/reference/method/db.collection.findOneAndUpdate/#std-label-findOneUpdate-returnNewDocument

Reply:
https://github.com/LangaraVoyagers/project-2-api/pull/8/commits/d50473c1beeff42ba45851c8cee9577f3cefb046